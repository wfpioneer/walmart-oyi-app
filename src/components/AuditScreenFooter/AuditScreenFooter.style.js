import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: COLOR.WHITE,
    justifyContent: 'space-around',
    flexDirection: 'row',
    flex: 1,
    paddingVertical: 5,
    paddingHorizontal: 10,
    width: '100%'
  },
  totalCntVw: {
    flex: 1,
    alignItems: 'center'
  },
  totalCnt: {
    fontWeight: '500',
    fontSize: 16
  },
  continueBtn: {
    flex: 1,
    marginHorizontal: 3
  }
});
