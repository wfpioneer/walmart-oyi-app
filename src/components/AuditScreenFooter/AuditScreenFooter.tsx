import React from 'react';
import { Text, View } from 'react-native';
import Button, { ButtonType } from '../buttons/Button';
import styles from './AuditScreenFooter.style';
import { strings } from '../../locales';

interface AuditScreenFooterProps {
  totalCount: number;
  onContinueClick: () => void;
  disabledContinue: boolean;
}

const AuditScreenFooter = ({ totalCount, onContinueClick, disabledContinue }: AuditScreenFooterProps) => (
  <View style={styles.container}>
    <View style={styles.totalCntVw}>
      <Text style={styles.totalCnt}>
        {`${totalCount} ${strings('GENERICS.ITEMS')}`}
      </Text>
    </View>
    <View style={styles.continueBtn}>
      <Button
        testID="btnContinue"
        title={strings('GENERICS.CONTINUE')}
        type={ButtonType.PRIMARY}
        onPress={onContinueClick}
        disabled={disabledContinue}
      />
    </View>
  </View>
);

export default AuditScreenFooter;
