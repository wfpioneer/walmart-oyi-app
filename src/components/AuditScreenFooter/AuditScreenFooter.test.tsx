import { fireEvent, render } from '@testing-library/react-native';
import AuditScreenFooter from './AuditScreenFooter';

describe('AuditScreenFooter Component', () => {
  it('Test renders AuditScreenFooter Screen with continue enabled', () => {
    const mockContinue = jest.fn();
    const { toJSON, getByTestId } = render(AuditScreenFooter({
      totalCount: 20, onContinueClick: mockContinue, disabledContinue: false
    }));

    const btnContinue = getByTestId('btnContinue');

    fireEvent.press(btnContinue);
    expect(mockContinue).toHaveBeenCalledWith();
    expect(toJSON()).toMatchSnapshot();
  });
  it('Test renders AuditScreenFooter Screen with continue disabled', () => {
    const mockContinue = jest.fn();
    const { toJSON } = render(AuditScreenFooter({
      totalCount: 20, onContinueClick: mockContinue, disabledContinue: true
    }));
    expect(toJSON()).toMatchSnapshot();
  });
});
