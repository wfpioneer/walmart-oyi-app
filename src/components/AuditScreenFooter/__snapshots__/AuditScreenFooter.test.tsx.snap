// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditScreenFooter Component Test renders AuditScreenFooter Screen with continue disabled 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      20 [missing "en.GENERICS.ITEMS" translation]
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 45,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5FA7F2",
              "borderRadius": 5,
              "elevation": 1,
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="btnContinue"
        >
          <Text
            style={
              Object {
                "color": "#ffffffff",
                "fontFamily": "Roboto-Medium",
                "fontSize": 16,
                "fontWeight": "normal",
              }
            }
          >
            [missing "en.GENERICS.CONTINUE" translation]
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`AuditScreenFooter Component Test renders AuditScreenFooter Screen with continue enabled 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      20 [missing "en.GENERICS.ITEMS" translation]
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 45,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#0071e9",
              "borderRadius": 5,
              "elevation": 1,
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="btnContinue"
        >
          <Text
            style={
              Object {
                "color": "#ffffffff",
                "fontFamily": "Roboto-Medium",
                "fontSize": 16,
                "fontWeight": "normal",
              }
            }
          >
            [missing "en.GENERICS.CONTINUE" translation]
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
